tests/test_kcca.py

```diff
@@ -126,4 +141,23 @@ def test_no_weights():
         kcca_b = KCCA(ktype ="linear", reg = 0.001, n_components = 1)
         kcca_b.transform([train1, train2])
         
-    
\ No newline at end of file
+# Test mrank error
+def test_mrank_neg():
+    with pytest.raises(ValueError):
+        kcca_z = KCCA(ktype ="poly", decomp = "icd", reg = 0.001, mrank = -1)
+        
```

Comment:
add test for mrank greater than n_components, since you said it must be less than

Reply:
actually doesnt need to be less than. only less than number of samples. realized this broke my code